everett/manager.py

```diff
@@ -136,15 +173,46 @@ def __init__(self, cfg):
         self.cfg = cfg
 
     def get(self, key, namespace=None):
-        if namespace:
-            key = list(namespace) + [key]
-            key = '_'.join(key)
+        return get_key_from_dict(self.cfg, key, namespace)
 
-        key = key.upper()
 
-        if key in self.cfg:
-            return self.cfg[key]
-        return NO_VALUE
+class ConfigEnvFileEnv(object):
+    """Source for pulling configuration out of .env files
+
+    This source lets you specify configuration in an .env file. This
+    is useful for local development when in production you use values
+    in environment variables.
+
+    Keys are prefixed by namespaces and the whole thing is uppercased.
+
+    For example, key "foo" will be ``FOO`` in the file.
+
+    For example, namespace "bar" for key "foo" becomes ``BAR_FOO`` in the
+    file.
+
+    Key and namespace can consist of alphanumeric characters and ``_``.
+
+    To use, instantiate and toss in the source list::
+
+        from everett.manager import ConfigEnvFileEnv, ConfigManager
+
+        config = ConfigManager([
+            ConfigEnvFileEnv('.env')
+        ])
+    """
+    data = None
+
+    def __init__(self, path):
+        path = os.path.abspath(os.path.expanduser(path.strip()))
+        if path and os.path.exists(path) and os.path.isfile(path):
```

Comment:
`os.path.isfile()` will both tell you if it exists as well as if it's a file, so you can skip the `os.path.exists(path)` check.

```
Python 2.7.12 (default, Jul  1 2016, 15:12:24) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import os.path
>>> os.path.isfile('foo')
False
>>> os.path.isfile('docker-compose.yml')
True
>>> os.path.isfile('dockerfiles')
False
>>> 
```

```
Python 3.5.2 (default, Jul  5 2016, 12:43:10) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os.path
>>> os.path.isfile('foo')
False
>>> os.path.isfile('docker-compose.yml')
True
>>> os.path.isfile('dockerfiles')
False
>>> 
```